core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -88,6 +88,10 @@ class AbstractModel:
         Hyperparameters that will be used by the model (can be search spaces instead of fixed values).
         If None, model defaults are used. This is identical to passing an empty dictionary.
     """
+    ag_key: str | None = None  # set to string value for subclasses for use in AutoGluon
+    ag_name: str | None = None  # set to string value for subclasses for use in AutoGluon
```

Comment:
I feel that both `ag_key` and `ag_name` serve a similar purpose, and one of them could potentially be removed. For example, we could generate the `ag_name` automatically from the class name with something like `self.name = re.sub(r"Model$", "", self.__class__.__name__)`.

Reply:
Agreed that they serve a similar purpose. For now I will keep both for legacy reasons, but we can consider standardizing. Maybe we can have `self.ag_name` default to your regix logic if it is `None`? Such as `self._ag_name = None` with def ag_name()` as a property?

Reply:
Note that AG Tabular doesn't have the same orderly name structure of the classes as TimeSeries (but I probably should do it soon...):

For example:

RFModel -> RandomForest
XTModel -> ExtraTrees
LGBModel -> LightGBM